crates/matrix-sdk-indexeddb/src/crypto_store/mod.rs

```diff
@@ -159,6 +159,53 @@ impl From<IndexeddbCryptoStoreError> for CryptoStoreError {
 
 type Result<A, E = IndexeddbCryptoStoreError> = std::result::Result<A, E>;
 
+/// Defines an operation to perform on the database
+enum DbOperation {
+    PutKeyVal { key: JsValue, value: JsValue },
+    Delete(JsValue),
+}
+/// A structs that represents all the operations that need to be done to the
+/// database when save_changes is called.
+/// The idea is to do all the serialization and encryption before the
+/// transaction, and then just do the actual Indexeddb operations in the
+/// transaction.
+struct IndexeddbChangesKeyValue {
+    /// A map of the object store names to the operations to perform on that
+    /// store.
+    store_to_key_values: HashMap<&'static str, Vec<DbOperation>>,
```

Comment:
nit: maybe prefer `BTreeMap` instead of `HashMap`? (iterating on BTreeMap is deterministic, which is nice during debugging sessions, and it's not the case for Hashmap)